lib/dialects/sqlite/query-generator.js

```diff
@@ -223,6 +223,74 @@ const QueryGenerator = {
     return `PRAGMA INDEX_LIST(${this.quoteTable(tableName)})`;
   },
 
+  getConstraintSnippet(tableName, options) {
+    let constraintSnippet, constraintName;
+    const fieldsSql = options.fields.map(field => {
+      if (typeof field === 'string') {
+        return this.quoteIdentifier(field);
+      } else if (field._isSequelizeMethod) {
+        return this.handleSequelizeMethod(field);
+      } else {
+        let result = '';
+
+        if (field.attribute) {
+          field.name = field.attribute;
+        }
+
+        if (!field.name) {
+          throw new Error('The following index field has no name: ' + field);
+        }
+
+        result += this.quoteIdentifier(field.name);
+        return result;
+      }
+    });
+    const fieldsSqlQuotedString = fieldsSql.join(', ');
+    const fieldsSqlString = fieldsSql.join('_');
+    switch (options.type.toUpperCase()) {
+      case 'UNIQUE':
+        constraintName = this.quoteIdentifier(options.name || `UK__${tableName}__${fieldsSqlString}`);      
+        constraintSnippet = `CONSTRAINT ${constraintName} UNIQUE (${fieldsSqlQuotedString})`;
```

Comment:
This code section is almost duplicate of `abstract/query-gen` `addConstraint` method, can we use that method / minimize code duplication ?